pkg/gce-pd-csi-driver/controller.go

```diff
@@ -590,9 +593,9 @@ func (gceCS *GCEControllerServer) executeControllerUnpublishVolume(ctx context.C
 	project, volKey, err = gceCS.CloudProvider.RepairUnderspecifiedVolumeKey(ctx, project, volKey)
 	if err != nil {
 		if gce.IsGCENotFoundError(err) {
-			return nil, status.Errorf(codes.NotFound, "ControllerUnpublishVolume could not find volume with ID %v: %v", volumeID, err)
+			return nil, status.Errorf(codes.NotFound, "ControllerUnpublishVolume could not find volume with ID %v: %v", volumeID, err.Error())
 		}
-		return nil, status.Errorf(codes.Internal, "ControllerUnpublishVolume error repairing underspecified volume key: %v", err)
+		return nil, LoggedError("ControllerUnpublishVolume error repairing underspecified volume key: ", err)
```

Comment:
Why are we logging only here, and not when the volume is not found? Probably we should log everything (?)

Reply:
We can add logging for notFound. Initially this pr is just fix unwrap error incorrectly. But at this point it is a blend of adding logs + fixing error unwrap.

Reply:
ok, fair enough.

Reply:
Okay i added the change for fixing status.Error to status.Errorf as well in this pr but committed it in a second commit. It looks much simpler :D